api/segments/views.py

```diff
@@ -42,9 +48,24 @@ def get_queryset(self):
 
         identity_pk = self.request.query_params.get("identity")
         if identity_pk:
-            identity = Identity.objects.get(pk=identity_pk)
-            queryset = queryset.filter(
-                id__in=[segment.id for segment in identity.get_segments()]
-            )
+            if identity_pk.isdigit():
+                identity = Identity.objects.get(pk=identity_pk)
+                segment_ids = [segment.id for segment in identity.get_segments()]
+            else:
+                segment_ids = _get_segment_ids_for_edge_identity(identity_pk)
+            queryset = queryset.filter(id__in=segment_ids)
 
         return queryset
+
+
+def _get_segment_ids_for_edge_identity(identity_pk: str) -> list:
```

Comment:
It feels like this method would be better off living somewhere else so we can unit test it separately. 

Could we add it directly to the `Identity` model? Or perhaps to a dynamo module / wrapper?

Reply:
Moved to dynamo identity wrapper